middleware/cors.go

```diff
@@ -102,6 +102,10 @@ func CORSWithConfig(config CORSConfig) echo.MiddlewareFunc {
 					allowOrigin = o
 					break
 				}
+				if o != "*" && equalScheme(origin, o) && isSubDomain(origin, o) {
```

Comment:
- I don't think you need `o != "*"`
- Make `equalScheme()` part of isSubdomain()`